src/com/amazon/corretto/crypto/provider/Utils.java

```diff
@@ -572,4 +574,61 @@ static <T> T requireNonNull(final T obj, final String message) {
   static String requireNonNullString(final String s, final String message) {
     return requireNonNull(s, message);
   }
+
+  /** Assumes a >= 0 && b >= 0 returns 1 if a > b; otherwise 0 */
+  static int oneIfGreaterOtherwiseZero(final int a, final int b) {
+    return (b - a) >>> INT_UNSIGNED_SHIFT;
+  }
+
+  /**
+   * All the bytes in the trusted buffer between trustedStart and trustedStart + trustedLen are
+   * examined if otherLen != 0. Please make sure the trusted array is the one that we trust.
+   *
+   * @param trusted
+   * @param trustedStart
+   * @param trustedLen
+   * @param other
+   * @param otherStart
+   * @param otherLen
+   * @return true iff all the bytes in the specified ranges are equal
+   */
+  static boolean constantEquality(
```

Comment:
Nitpick: If this function is intended to be constant time, then it should be reflected in the name. Could this function be renamed to `constantTimeEquality` or `constantTimeEquals`.

Reply:
I'll rename in the next patch.